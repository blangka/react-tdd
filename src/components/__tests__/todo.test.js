import { render, screen, cleanup} from "@testing-library/react";
import Todo from "../todo";
import {createRenderer} from "react-dom/test-utils";
import renderer from "react-test-renderer";


afterEach(() => {
    cleanup();
});

test('test init', () => {
    expect(true).toBe(true);
})


test('should render todo compoenent', () => {

    const todos = [{
        id: 1,
        title: 'wash dished',
        completed: false
    },
        {
            id: 2,
            title: 'make dinner',
            completed: true
        }];
    render(<Todo todo={todos}/>);
    const todoElement = screen.getByTestId('todo-check');

    expect(todoElement).toBeInTheDocument();
    expect(todoElement).toHaveTextContent('Todo');
})

test('should render non-completed todo ', () => {

    const todo = {
        id: 1,
        title: 'wash dished',
        completed: false
    };
    render(<Todo todo={todo}/>);
    const todoElement = screen.getByTestId('todo-1');

    expect(todoElement).toBeInTheDocument();
    expect(todoElement).toHaveTextContent('wash dished');
    expect(todoElement).not.toContainHTML('strike');
})

test('should render completed todo ', () => {

    const todo =
        {
            id: 2,
            title: 'make dinner',
            completed: true
        };
    render(<Todo todo={todo}/>);
    const todoElement = screen.getByTestId('todo-2');

    expect(todoElement).toBeInTheDocument();
    expect(todoElement).toHaveTextContent('make dinner');
    expect(todoElement).toContainHTML('strike');
})

test('matches snapshot', () => {
    const todo = {
        id: 1,
        title: 'wash dished',
        completed: false
    };
    const tree = renderer.create(<Todo todo={todo}/>).toJSON();
    console.log(tree);
    expect(tree).toMatchSnapshot();
});
