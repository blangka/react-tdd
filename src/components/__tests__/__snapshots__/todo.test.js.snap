// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="todo-1"
>
  <h1
    data-testid="todo-check"
  >
    Todo
  </h1>
  <h1>
    wash dished
  </h1>
</div>
`;
